src/stdx.zig

```diff
@@ -745,3 +745,50 @@ pub fn EnumUnionType(
 pub fn comptime_slice(comptime slice: anytype, comptime len: usize) []const @TypeOf(slice[0]) {
     return &@as([len]@TypeOf(slice[0]), slice[0..len].*);
 }
+
+/// Return a Formatter for a u64 value representing a file size.
+/// This formatter statically checks that the number is a multiple of 1024,
+/// and represents it using the IEC measurement units (KiB, MiB, GiB, ...).
+pub fn fmt_int_size_bin_exact(comptime value: u64) std.fmt.Formatter(format_int_size_bin_exact) {
+    comptime assert(value % 1024 == 0);
+    return .{ .data = value };
+}
+
+fn format_int_size_bin_exact(
+    value: u64,
+    comptime fmt: []const u8,
+    options: std.fmt.FormatOptions,
+    writer: anytype,
+) !void {
+    _ = fmt;
+    if (value == 0) {
+        return std.fmt.formatBuf("0B", options, writer);
+    }
+
+    // The worst case in terms of space needed is 20 bytes,
+    // since `maxInt(u64)` is the highest number,
+    // + 3 bytes for the measurement units suffix.
+    var buf: [23]u8 = undefined;
+
```

Comment:
could you add `comptime assert` that comptime printing `maxInt(u64) GiB` would be not longer than 23?